models/aether_lat-lon/model_mod.f90

```diff
@@ -0,0 +1,616 @@
+! Read dimension information from the template file and use 
+! it to assign values to variables.
+
+subroutine assign_dimensions()
+
+integer  :: ncid
+character(len=24), parameter :: routine = 'assign_dimensions'
+
+call error_handler(E_MSG, routine, 'reading filter input ['//trim(template_file)//']')
+
+ncid = nc_open_file_readonly(template_file, routine)
+
+! levels
+nlev = nc_get_dimension_size(ncid, trim(LEV_DIM_NAME), routine)
+allocate(levs(nlev))
+call nc_get_variable(ncid, trim(LEV_VAR_NAME), levs, routine)
+
+! latitiude
+nlat = nc_get_dimension_size(ncid, trim(LAT_DIM_NAME), routine)
+allocate(lats(nlat))
+call nc_get_variable(ncid, trim(LAT_VAR_NAME), lats, routine)
+
+! longitude
+nlon = nc_get_dimension_size(ncid, trim(LON_DIM_NAME), routine)
+allocate(lons(nlon))
+call nc_get_variable(ncid, trim(LON_VAR_NAME), lons, routine)
+
+end subroutine assign_dimensions
+
+!-----------------------------------------------------------------------
+! Parse the table of variables' characteristics into arrays for easier access.
+
+function assign_var(variables, MAX_STATE_VARIABLES) result(var)
+
+   character(len=vtablenamelength), intent(in) :: variables(:, :)
+   integer, intent(in)                         :: MAX_STATE_VARIABLES
+
+   type(var_type) :: var
+
+   integer        :: ivar
+
+   !-----------------------------------------------------------------------
+   ! Codes for interpreting the NUM_STATE_TABLE_COLUMNS of the variables table
+   integer, parameter :: NAME_INDEX      = 1 ! ... variable name
+   integer, parameter :: QTY_INDEX       = 2 ! ... DART kind
+   integer, parameter :: MIN_VAL_INDEX   = 3 ! ... minimum value if any
+   integer, parameter :: MAX_VAL_INDEX   = 4 ! ... maximum value if any
+   integer, parameter :: UPDATE_INDEX    = 5 ! ... update (state) or not
+
+   ! Loop through the variables array to get the actual count of the number of variables
+   do ivar = 1, MAX_STATE_VARIABLES
+      ! If the element is an empty string, the loop has exceeded the extent of the variables
+      if (variables(1, ivar) == '') then
+          var%count = ivar-1
+          exit
+      endif 
+   enddo
+  
+   ! Allocate the arrays in the var derived type
+   allocate(var%names(var%count), var%qtys(var%count), var%clamp_values(var%count, 2), var%updates(var%count))
+  
+   do ivar = 1, var%count
+   
+      var%names(ivar) = trim(variables(NAME_INDEX, ivar))
+  
+      var%qtys(ivar) = get_index_for_quantity(variables(QTY_INDEX, ivar))
+  
+      if (variables(MIN_VAL_INDEX, ivar) /= 'NA') then
+          read(variables(MIN_VAL_INDEX, ivar), '(d16.8)') var%clamp_values(ivar,1)
+      else
+          var%clamp_values(ivar,1) = MISSING_R8
+      endif
+  
+      if (variables(MAX_VAL_INDEX, ivar) /= 'NA') then
+          read(variables(MAX_VAL_INDEX, ivar), '(d16.8)') var%clamp_values(ivar,2)
+      else
+          var%clamp_values(ivar,2) = MISSING_R8
+      endif
+  
+      if (variables(UPDATE_INDEX, ivar) == 'UPDATE') then
```

Comment:
case sensitve, so 'update' in the input.nml would give you var%updates(ivar) = .false.

Reply:
Added subroutine call to_upper(table_entry) before the var%updates element is assigned.